drivers/clock_control/clock_stm32_ll_mco.h

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright (c) 2017-2022 Linaro Limited.
+ * Copyright (c) 2017 RnDity Sp. z o.o.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZEPHYR_DRIVERS_CLOCK_CONTROL_CLOCK_STM32_LL_MCO_H_
+#define ZEPHYR_DRIVERS_CLOCK_CONTROL_CLOCK_STM32_LL_MCO_H_
+
+#include <stm32_ll_utils.h>
+
+#if CONFIG_CLOCK_STM32_MCO1_SRC_NOCLOCK
+	#define MCO1_SOURCE		LL_RCC_MCO1SOURCE_NOCLOCK
+#elif CONFIG_CLOCK_STM32_MCO1_SRC_EXT_HSE
+	#define MCO1_SOURCE		LL_RCC_MCO1SOURCE_EXT_HSE
+#elif CONFIG_CLOCK_STM32_MCO1_SRC_LSE
+	#define MCO1_SOURCE		LL_RCC_MCO1SOURCE_LSE
+#elif CONFIG_CLOCK_STM32_MCO1_SRC_HSE
+	#define MCO1_SOURCE		LL_RCC_MCO1SOURCE_HSE
+#elif CONFIG_CLOCK_STM32_MCO1_SRC_LSI
+	#define MCO1_SOURCE		LL_RCC_MCO1SOURCE_LSI
+#elif CONFIG_CLOCK_STM32_MCO1_SRC_MSI
+	#define MCO1_SOURCE		LL_RCC_MCO1SOURCE_MSI
+#elif CONFIG_CLOCK_STM32_MCO1_SRC_HSI
+	#define MCO1_SOURCE		LL_RCC_MCO1SOURCE_HSI
+#elif CONFIG_CLOCK_STM32_MCO1_SRC_HSI16
+	#define MCO1_SOURCE		LL_RCC_MCO1SOURCE_HSI
+#elif CONFIG_CLOCK_STM32_MCO1_SRC_HSI48
+	#define MCO1_SOURCE		LL_RCC_MCO1SOURCE_HSI48
+#elif CONFIG_CLOCK_STM32_MCO1_SRC_PLLCLK
+	#define MCO1_SOURCE		LL_RCC_MCO1SOURCE_PLLCLK
+#elif CONFIG_CLOCK_STM32_MCO1_SRC_PLLQCLK
+	#define MCO1_SOURCE		LL_RCC_MCO1SOURCE_PLL1QCLK
```

Comment:
The `CONFIG_CLOCK_STM32_MCO1_SRC_PLLQCLK `could introduce a confusion : 
- stm32H5X and stm32H7xX have LL_RCC_MCO1SOURCE_PLL1QCLK
- stm32G0X and stm32WLX have LL_RCC_MCO1SOURCE_PLLQCLK

Reply:
I named it this way because "PLL1" is named "PLL" but I didn't think about the potential name conflict actually. Thank you.
I see 2 solutions:
- renaming CONFIG_CLOCK_STM32_MCO1_SRC_PLLQCLK into CONFIG_CLOCK_STM32_MCO1_SRC_PLL1QCLK
- checking against SOC_STM32_SERIES_XX to choose between LL_RCC_MCO1SOURCE_PLL1QCLK or LL_RCC_MCO1SOURCE_PLLQCLK
Which one seems best to you ?

Reply:
I'd prefer we avoid the first option. PLL and PLL1 are just different naming for main PLL. 2nd option is fine to me

Reply:
Fixed !